crates/revm/src/interpreter.rs

```diff
@@ -123,6 +123,18 @@ impl Interpreter {
             self.program_counter = unsafe { self.program_counter.offset(1) };
             ret = eval::<H, SPEC>(opcode, self, host);
 
+            // CALL failures don't fail current call by themselves.
+            if matches!(
+                opcode,
+                super::instructions::opcode::CALL
+                    | super::instructions::opcode::STATICCALL
+                    | super::instructions::opcode::DELEGATECALL
+            ) {
+                if !matches!(ret, return_ok!() | Return::OutOfGas) {
+                    ret = Return::Continue
+                }
+            }
```

Comment:
This loop is hot part of evm, as it is an execution loop and these checks does not seem right.

Reply:
> This loop is hot part of evm, as it is an execution loop and these checks does not seem right.

Do you mean checking on each step if opcode was `CALL | DELEGATECALL | STATICCALL` might slow things down?

> This seems not correct if ret is Return::OutOfGas we can't just override that to Return::Continue

We do not override `Return::OutOfGas` to `Return::Continue`. We only override when it is neither of `return_ok!() | Return::OutOfGas` to `Return::Continue` to keep the behaviour same as earlier https://github.com/bluealloy/revm/pull/136/files#diff-6e6245d0a2d234f8bc8ea76eefc437a171d3dfc1fe97098603966161ac491244L403 where we only ever returned `Return::Continue` or `Return::OutOfGas`(from `next_gas_block` checks). Execution was not halted on anything other than `OutOfGas`. Here, we are just trying to bubble up the actual reason / status of `CALL` evaluation, but it still shouldn't be halted unless it is out of gas.

Reply:
Yes, it will slow it down.
Yeah i missed `!`  in `if !matches!(ret,...` logic is the same.

I am not seeing who is using this? Additional inspector calls for `call_end` are okay and should be added, but this code is called right after `CALL | CALLCODE | DELEGATECALL | STATICCALL` ( you are missing CALLCODE) and it does nothing, or I am missing something?

Reply:
Whoops, you are right. I was thinking of providing the status to `step_end`, I need to move the `step_end` call above this `if`. However, thinking about it now, I think we'd need to change the interface a bit to give the correct status(the overriden status) vs intermediate status(call's return) since by default an inspector does nothing and returns the same status that it was given. If we replace the current status with intermediate status, we'd trigger termination.

Will think about this more whether we even need to know the intermediate status of `CALL` step because we can use `call_end` to record the same in the inspector.